### PR TITLE
Implement keychain access & refresh token on startup

### PR DESCRIPTION
**Pull request by FrancoisDevemy5**
[[Sandbox] Add an example of session persistance with RT](https://app.asana.com/0/1108828962586858/1192984719275649/f)
Store the auth token in the secure keychain.
Use the refresh token to get a new auth token when coming back to the fore.

### CODE REVIEW

File: Sandbox/Sandbox/Base.lproj/Main.storyboard [hunk 1]
```diff
@@ -154,7 +154,7 @@
                                 <textInputTraits key="textInputTraits"/>
                             </textField>
                             <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Redirect URI" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="yXF-2L-fNb">
-                                <rect key="frame" x="20" y="102" width="95.5" height="21"/>
+                                <rect key="frame" x="20" y="102" width="95.666666666666686" height="21"/>
```
Comment (roxanemace) [inline Sandbox/Sandbox/Base.lproj/Main.storyboard:157]: Why don't you round the number?

File: Sandbox/Sandbox/AppDelegate.swift [hunk 2]
```diff
@@ -7,75 +7,106 @@ import IdentitySdkGoogle
...
     func applicationDidEnterBackground(_ application: UIApplication) {
+        print("applicationDidEnterBackground")
         // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
         // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
     }
 
     func applicationWillEnterForeground(_ application: UIApplication) {
         // Called as part of the transition from the background to the active state; here you can undo many of the changes made on entering the background.
+        print("applicationWillEnterForeground")
+        guard let token: AuthToken = AppDelegate.storage.get(key: "AUTH_TOKEN") else {
```
Comment (roxanemace) [inline Sandbox/Sandbox/AppDelegate.swift:74]: Maybe you can create a variable for the `AUTH_TOKEN` key?

**Review by roxanemace — APPROVED** [review]

File: Sandbox/Sandbox/Base.lproj/Main.storyboard (hunk 1 above)
Comment (FrancoisDevemy5) [reply Sandbox/Sandbox/Base.lproj/Main.storyboard:157]: I don't know what this was changed. I don't control the storyboard directly. Maybe I should just revert this change...

Comment (roxanemace) [reply Sandbox/Sandbox/Base.lproj/Main.storyboard:157]: It's ok, if XCode did it, then it must have a good reason.

Comment (FrancoisDevemy5) [reply Sandbox/Sandbox/Base.lproj/Main.storyboard]: I reverted and saw no difference. I pushed the revert.

File: Sandbox/Sandbox/AppDelegate.swift [hunk 3]
```diff
@@ -7,75 +7,107 @@ import IdentitySdkGoogle
...
-    
+
     func applicationWillResignActive(_ application: UIApplication) {
+        print("applicationWillResignActive")
         // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
         // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
     }
 
     func applicationDidEnterBackground(_ application: UIApplication) {
+        print("applicationDidEnterBackground")
```
Comment (axuan25) [inline Sandbox/Sandbox/AppDelegate.swift:67]: Do we still want the print statements for the sandbox?

**Review by axuan25 — APPROVED** [review]

File: Sandbox/Sandbox/AppDelegate.swift (hunk 3 above)
Comment (FrancoisDevemy5) [reply Sandbox/Sandbox/AppDelegate.swift:67]: This sandbox is for education purpose for now. And we still haven't reactivated the Alamofire logging.

Comment (FrancoisDevemy5) [reply Sandbox/Sandbox/AppDelegate.swift:67]: So not ideal, but it is helpful for now.

**Pull request merged**